### DISCUSSION TITLE
Tectonic at TUG2022!

### DISCUSSION THREAD

**Discussion by pkgw** [tectonic-typesetting/tectonic#886] [Announcements]
Hi folks — I'm delighted to report that I've been invited to deliver a keynote about Tectonic at [TUG 2022][x]! It will be a virtual conference taking place July 22–24, 2022. Below are the title and abstract that I've drafted.

[x]: https://tug.org/tug2022/

I suspect that I'll have absolutely no shortage of material to cover in my presentation, but please let me know if there are particular topics that you'd like to see me discuss!

---

### The Tectonic Project: Envisioning A 21st-Century TeX Experience

Tectonic is a software project built around an alternative TeX engine forked from XeTeX. It was created to explore the answers to two questions. The first question relates to documents: in a world of 21st-century technologies — where interactive displays, computation, and internet connectivity are generally cheap and ubiquitous — what new forms of technical document have become possible? The second question relates to tools: how can we use those same technologies to do a better job of empowering people to create excellent technical documents? The answers are, of course, intertwined: without a system of great tools, it's hard (or perhaps impossible?) to create great documents.

The premises of the Tectonic project are that the world needs and deserves a "21st-century" document authoring system, that such a system should have TeX at its heart — and that in order to create a successful system, parts of the classic TeX experience will need to be rethought or jettisoned completely. This is why Tectonic forks XeTeX and is branded independently: while it aspires to maintain compatibility with classic TeX workflows as far as can be managed, in a certain sense the whole point of the effort is to break compatibility and ignore tradition — to experiment with new ideas that can't be tried in mainline TeX. Thus far, these "new ideas" have focused on experience design, seeking to deliver a system that is convenient, empowering, and even delightful for users and developers. Tectonic is therefore compiled using standard Rust tools, installs as a single executable file, and downloads support files from a prebuilt TeXLive distribution on demand. In the past year, long-threatened work on native HTML output has finally started landing, including a possibly novel Unicode math rendering scheme based on font subsetting. The goal for upcoming work is to flesh out this HTML support so that Tectonic can create the world's best web-native technical documents, and to use that support to document the Tectonic system itself.

**Comment by ratmice** [comment]
> I suspect that I'll have absolutely no shortage of material to cover in my presentation, but please let me know if there are particular topics that you'd like to see me discuss!

> Tectonic is therefore compiled using standard Rust tools, installs as a single executable file

One thing not mentioned above is the usability of tectonic as a library, and there are a few users mostly in the area of code documentation generators.  On the 21st-century documents, one of my goals (unrealized) has been interactive editing of documentation for web-like or rustdoc like comment systems with inline rendering of documentation, allowing you to switch between editing and rendered documentation seamlessly, as opposed to viewing documentation as a separate artifact.

**Comment by pkgw** [comment]
>  On the 21st-century documents, one of my goals (unrealized) has been interactive editing of documentation for web-like or rustdoc like comment systems with inline rendering of documentation, allowing you to switch between editing and rendered documentation seamlessly, as opposed to viewing documentation as a separate artifact.

This may not be focusing on quite the same thing as you, but speaking of interactive editing, I really hope that one day the backbone of the Tectonic documentation will be a sort of "Tectonopedia" of definitions and reference information, hopefully arising with a lot of organic contributions à la, yes, Wikipedia. I don't think that longer narrative documentation necessarily lends itself to crowdsourcing so well, but often when I'm trying to write such documentation I wish I had a sort of Tectonic-specific Wikipedia to link to, and empirically that's the sort of thing that crowdsources well! But I think you'd need good interactive editing tools to help make that happen.

(Here I'm being influenced by ["the" documentation system][tds] and its model of four primary kinds of documentation, where the Tectonopedia would provide the reference kind.)

[tds]: https://documentation.divio.com/

**Comment by pkgw** [comment]
Also, for a bit more depth on what the abstract is getting at, I recently reread an item that I wrote about Tectonic in 2018 and I was like "hey, that's pretty well-put!": [(Belatedly) Introducing Tectonic](https://newton.cx/~peter/2018/belatedly-introducing-tectonic/).
